Some refactoring of website php

lonvia opened the pull request:
* introduce accessor function for URL parameter
  * remove unused `bAsPoints` from reverse.php
  * update links to OSM objects (\browse\ no longer necessary, allow https)
  * factor out link formatting function and osm type translation
  * introduce chksql function for website code

Most important for the user-facing API is the more strict checking of input parameters. Illegal value now result in a 400 error instead of simply returning nothing.

mtmail commented on lib/website.php:
```diff
@@ -0,0 +1,140 @@
+			echo "<pre>\n".$oSql->getUserInfo()."</pre>";
+		}
+
+		echo "</pre></p></body></head></html>";
```
closing </head> misplaced here.

mtmail commented on lib/website.php:
```diff
@@ -0,0 +1,140 @@
+		header('Content-type: text/html; charset=utf-8');
+		echo "<html><body><h1>Internal Server Error</h1>";
+		echo '<p>Nominatim has encountered an internal error while processing your request. This is most likely because of a bug in the software.</p>';
+		echo "<p><b>Details:</b> ".$sError,"</p>";
```
should the comma be a dot in this case?

lonvia commented:
@mtmail Thanks, fixed.

I've also added a fix where the reverse HTML would not work anymore when zoom was not set. My error checking was a bit too strict.

mtmail commented:
Excellent refactoring. trac replaced with github URLs, no longer using a `_GET` value in left-assignment, lots of tiny good changes as well.
